host/tests/dsClasses/dsVideoPort.py

```diff
@@ -0,0 +1,1465 @@
+        portLists = []
+
+        ports = self.deviceProfile.get("Ports")
+        for i in range(1, len(ports)+1):
+            entry = ports[i]
+            portLists.append([dsVideoPortType(entry['Typeid']).name, entry['Index']])
+
+        return portLists
+
+    def getHDCPVersion(self):
+        """
+        Returns the supported HDCP on device.
+
+        Args:
+            None.
+
+        Returns:
+            returns the supported hdcp_protocol_version
+        """
+        hdcp_protocol_version = 0
+
+        ports = self.deviceProfile.get("Ports")
+        for i in range(1, len(ports)+1):
+            entry = ports[i]
+            hdcp_protocol_version = dsVideoPorthdcp_version(entry['hdcp_protocol_version']).name
+
+        return hdcp_protocol_version
+
+    def getResolutions(self):
+        """
+        Returns the supported Resolutions on device.
+
+        Args:
+            None.
+
+        Returns:
+            returns the supported resolutions
+        """
+        # Store the resolutions in a list
+        resolutions_list = []
+        pixel_res = 0
+
+        ports = self.deviceProfile.get("Ports")
+        for i in range(1, len(ports)+1):
+            entry = ports[i]
+            num_supported_resolutions = entry["numSupportedResolutions"]
+            supported_resolutions = entry["supportedResolutions"]
+            for j in range(1, num_supported_resolutions + 1):
+                resolutions_data = supported_resolutions.get(j)
+                if resolutions_data:
+                    # Convert pixelResolution value to dsVideoResolution enum and add to list
+                    pixel_res = resolutions_data.get("pixelResolution")
+                    resolution_enum = dsVideoResolution(pixel_res).name
+                    aspectRatio_res = resolutions_data.get("aspectRatio")
+                    aspectRatio_enum = dsVideoAspectRatio(aspectRatio_res).name
+                    stereoScopicMode_res = resolutions_data.get("stereoScopicMode")
+                    stereoScopicMode_enum = dsVideoStereoScopicMode(stereoScopicMode_res).name
+                    frameRate_res = resolutions_data.get("frameRate")
+                    frameRate_enum = dsVideoFrameRate(frameRate_res).name
+                    scanModes_res = resolutions_data.get("interlaced")
+                    scanModes_enum = dsVideoScanModeMode(scanModes_res).name
+
+                    resolutions_list.append({
+                        "pixelResolution":resolution_enum,
+                        "aspectRatio":aspectRatio_enum,
+                        "stereoScopicMode":stereoScopicMode_enum,
+                        "frameRate":frameRate_enum,
+                         "interlaced":scanModes_enum
+                        })
+
+        return resolutions_list
+
+    def getAspectRatio(self):
+        """
+        Returns the supported aspectRatio on device.
+
+        Args:
+            None.
+
+        Returns:
+            returns the supported aspectRatio
```

Comment:
you need to be specific about what format the returned data is in, and give an example.